MonkeyWrench.Web.WebService/Scheduler/Scheduler.cs

```diff
@@ -400,11 +396,69 @@ private static void AddWork (DB db, List<DBHost> hosts, List<DBLane> lanes, List
 								}
 							}
 
-							sql.AppendFormat ("UPDATE RevisionWork SET state = {0} WHERE id = {1} AND state = 10;", (int) (has_dependencies ? DBState.DependencyNotFulfilled : DBState.NotDone), revisionwork.id);
+							sql.AppendFormat ("UPDATE RevisionWork SET state = 11 WHERE id = {0} AND state = 10;", revisionwork.id);
+							lines++;
+							if (!editedHostLanes.Contains (hostlane)) editedHostLanes.Add (hostlane);
+						}
+					}
+				}
+
+				if (editedHostLanes.Count > 0) {
+					using (IDbCommand cmd = db.CreateCommand ()) {
+						bool first = true;
+						int revisionwork_id;
+						int lane_id;
+						int state;
 
+						cmd.CommandText = @"
+SELECT RevisionWork.id, RevisionWork.lane_id
+FROM RevisionWork
+INNER JOIN Revision ON Revision.id = RevisionWork.revision_id
+INNER JOIN (
+	SELECT a.lane_id, a.date, a.id
+	FROM Revision a
+	INNER JOIN (
+		SELECT lane_id, MAX (date) DATE
+		FROM Revision
+		GROUP BY lane_id
+	) b ON a.lane_id = b.lane_id AND a.date = b.date
+	INNER JOIN Lane ON Lane.id = a.lane_id
+	WHERE Lane.enabled = TRUE
+		AND (
+";
+
+						foreach (var editedHostLane in editedHostLanes) {
+							if (first) {
+								cmd.CommandText += String.Format ("Lane.id = {0}", editedHostLane.lane_id);
+								first = false;
+							} else {
+								cmd.CommandText += String.Format ("OR Lane.id = {0}", editedHostLane.lane_id);
+							}
+						}
+						cmd.CommandText += ")) LATEST ON RevisionWork.lane_id = latest.lane_id AND RevisionWork.revision_id = latest.id";
+
+						using (IDataReader reader = cmd.ExecuteReader ()) {
+							while (reader.Read ()) {
+								if ((lines % 100) == 0 || (lines % 101) == 0) {
+									db.ExecuteNonQuery (sql.ToString ());
+									sql.Clear ();
+									log.DebugFormat ("AddWork: flushed work queue, added {0} items now.", lines);
+								}
+								revisionwork_id = reader.GetInt32 (0);
+								lane_id = reader.GetInt32 (1);
+
+								has_dependencies = dependencies != null && dependencies.Any (dep => dep.lane_id == lane_id);
+								state = (int)(has_dependencies ? DBState.DependencyNotFulfilled : DBState.NotDone);
+								log.DebugFormat ("AddWork: Lane '{0}', revisionwork_id '{1}' has dependencies: {2}", lane_id, revisionwork_id, has_dependencies);
+
+								sql.AppendFormat ("UPDATE Revisionwork SET state = {0} WHERE id = {1} AND state = 11;", state, revisionwork_id);
+								sql.AppendFormat ("UPDATE Work SET state = {0} WHERE revisionwork_id = {1} AND state = 11;", state, revisionwork_id);
+								lines += 2;
+							}

```

Comment:
Isn't that taken care of by lines 462/463 https://github.com/mono/monkeywrench/pull/107/files/b0c3d35bcd91fedfd06c75851319e62d4fa49b75#diff-a89067d97cb28394f58edece0b3a65acR463?